File: python/helpers/dataGeneration.py
from __future__ import print_function, division
import numpy as np
from numpy.random import randn

# Generates data from the LGSS model
def generateData(theta, noObservations, initialState):

    phi = theta[0]
    sigmav = theta[1]
    sigmae = theta[2]

    state = np.zeros(noObservations + 1)
    observation = np.zeros(noObservations)
    state[0] = initialState

    for t in range(1, noObservations):
        state[t] = phi * state[t - 1] + sigmav * randn()
        observation[t] = state[t] + sigmae * randn()

    return(state, observation)

File: r/helpers/dataGeneration.R
##############################################################################
# Generating data from a LGSS model
#
# Johan Dahlin <uni (at) johandahlin.com.nospam>
# Documentation at https://github.com/compops/pmh-tutorial
# Published under GNU General Public License
##############################################################################
generateData <- function(theta, noObservations, initialState)
{
  phi <- theta[1]
  sigmav <- theta[2]
  sigmae <- theta[3]

  state <- matrix(0, nrow = noObservations + 1, ncol = 1)
  observation <- matrix(0, nrow = noObservations + 1, ncol = 1)

  state[1] <- initialState
  observation[1] <- NA

  for (t in 2:(noObservations + 1)) {
    state[t] <- phi * state[t - 1] + sigmav * rnorm(1)
    observation[t] <- state[t] + sigmae * rnorm(1)
  }

  list(x = state, y = observation)
}